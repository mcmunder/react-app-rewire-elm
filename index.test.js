const config = require('react-scripts/config/webpack.config.dev')

const {rewireElm} = require('./index.js')

describe('rewireElm', () => {
  test('produces expected webpack config', () => {
    expect(rewireElm(config)).toMatchSnapshot()
  })
})
