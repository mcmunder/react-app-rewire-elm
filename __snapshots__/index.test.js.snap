// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rewireElm produces expected webpack config 1`] = `
Object {
  "devtool": "cheap-module-source-map",
  "entry": Array [
    "/Users/mcmunder/code/react-app-rewire-elm/node_modules/react-scripts/config/polyfills.js",
    "/Users/mcmunder/code/react-app-rewire-elm/node_modules/react-dev-utils/webpackHotDevClient.js",
    "/Users/mcmunder/code/react-app-rewire-elm/src/index.js",
  ],
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "pre",
        "include": "/Users/mcmunder/code/react-app-rewire-elm/src",
        "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
        "use": Array [
          Object {
            "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/eslint-loader/index.js",
            "options": Object {
              "baseConfig": Object {
                "extends": Array [
                  "/Users/mcmunder/code/react-app-rewire-elm/node_modules/eslint-config-react-app/index.js",
                ],
              },
              "eslintPath": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/react-scripts/node_modules/eslint/lib/api.js",
              "formatter": [Function],
              "ignore": false,
              "useEslintrc": false,
            },
          },
        ],
      },
      Object {
        "oneOf": Array [
          Object {
            "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/url-loader/index.js",
            "options": Object {
              "limit": 10000,
              "name": "static/media/[name].[hash:8].[ext]",
            },
            "test": Array [
              /\\\\\\.bmp\\$/,
              /\\\\\\.gif\\$/,
              /\\\\\\.jpe\\?g\\$/,
              /\\\\\\.png\\$/,
            ],
          },
          Object {
            "include": "/Users/mcmunder/code/react-app-rewire-elm/src",
            "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/babel-loader/lib/index.js",
            "options": Object {
              "babelrc": false,
              "cacheDirectory": true,
              "presets": Array [
                "/Users/mcmunder/code/react-app-rewire-elm/node_modules/babel-preset-react-app/index.js",
              ],
            },
            "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
          },
          Object {
            "test": /\\\\\\.css\\$/,
            "use": Array [
              "/Users/mcmunder/code/react-app-rewire-elm/node_modules/style-loader/index.js",
              Object {
                "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/css-loader/index.js",
                "options": Object {
                  "importLoaders": 1,
                },
              },
              Object {
                "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/postcss-loader/lib/index.js",
                "options": Object {
                  "ident": "postcss",
                  "plugins": [Function],
                },
              },
            ],
          },
          Object {
            "exclude": Array [
              /node_modules/,
            ],
            "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/elm-webpack-loader/index.js",
            "test": /\\\\\\.elm\\$/,
          },
          Object {
            "exclude": Array [
              /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
              /\\\\\\.html\\$/,
              /\\\\\\.json\\$/,
              "/.elm$/",
            ],
            "loader": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/file-loader/dist/cjs.js",
            "options": Object {
              "name": "static/media/[name].[hash:8].[ext]",
            },
          },
        ],
      },
    ],
    "strictExportPresence": true,
  },
  "node": Object {
    "child_process": "empty",
    "dgram": "empty",
    "fs": "empty",
    "net": "empty",
    "tls": "empty",
  },
  "output": Object {
    "chunkFilename": "static/js/[name].chunk.js",
    "devtoolModuleFilenameTemplate": [Function],
    "filename": "static/js/bundle.js",
    "pathinfo": true,
    "publicPath": "/",
  },
  "performance": Object {
    "hints": false,
  },
  "plugins": Array [
    InterpolateHtmlPlugin {
      "replacements": Object {
        "NODE_ENV": "test",
        "PUBLIC_URL": "",
      },
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": false,
        "inject": true,
        "minify": false,
        "showErrors": true,
        "template": "/Users/mcmunder/code/react-app-rewire-elm/public/index.html",
        "title": "Webpack App",
        "xhtml": false,
      },
    },
    NamedModulesPlugin {
      "options": Object {},
    },
    DefinePlugin {
      "definitions": Object {
        "process.env": Object {
          "NODE_ENV": "\\"test\\"",
          "PUBLIC_URL": "\\"\\"",
        },
      },
    },
    HotModuleReplacementPlugin {
      "fullBuildTimeout": 200,
      "multiStep": undefined,
      "options": Object {},
      "requestTimeout": 10000,
    },
    CaseSensitivePathsPlugin {
      "fsOperations": 0,
      "options": Object {},
      "pathCache": Object {},
      "primed": false,
    },
    WatchMissingNodeModulesPlugin {
      "nodeModulesPath": "/Users/mcmunder/code/react-app-rewire-elm/node_modules",
    },
    IgnorePlugin {
      "checkIgnore": [Function],
      "contextRegExp": /moment\\$/,
      "resourceRegExp": /\\^\\\\\\.\\\\/locale\\$/,
    },
  ],
  "resolve": Object {
    "alias": Object {
      "babel-runtime": "/Users/mcmunder/code/react-app-rewire-elm/node_modules/babel-runtime",
      "react-native": "react-native-web",
    },
    "extensions": Array [
      ".web.js",
      ".mjs",
      ".js",
      ".json",
      ".web.jsx",
      ".jsx",
      ".elm",
    ],
    "modules": Array [
      "node_modules",
      "/Users/mcmunder/code/react-app-rewire-elm/node_modules",
    ],
    "plugins": Array [
      ModuleScopePlugin {
        "allowedFiles": Set {
          "/Users/mcmunder/code/react-app-rewire-elm/package.json",
        },
        "appSrc": "/Users/mcmunder/code/react-app-rewire-elm/src",
      },
    ],
  },
}
`;
